block-storage/src/main/scala/coop/rchain/blockstorage/deploy/KeyValueDeployStorage.scala

```diff
@@ -0,0 +1,33 @@
+package coop.rchain.blockstorage.deploy
+
+import cats.Functor
+import cats.effect.Sync
+import cats.syntax.all._
+import com.google.protobuf.ByteString
+import coop.rchain.blockstorage.dag.codecs.{codecDeploySignature, codecSignedDeployData}
+import coop.rchain.casper.protocol.DeployData
+import coop.rchain.crypto.signatures.Signed
+import coop.rchain.shared.syntax._
+import coop.rchain.store.{KeyValueStoreManager, KeyValueTypedStore}
+
+final case class KeyValueDeployStorage[F[_]: Functor] private (
+    store: KeyValueTypedStore[F, ByteString, Signed[DeployData]]
+) extends DeployStorage[F] {
+  def add(deploys: List[Signed[DeployData]]): F[Unit] =
+    store.put(deploys.map(d => (d.sig, d)))
+
+  def remove(deploys: List[Signed[DeployData]]): F[Int] =
+    store.delete(deploys.map(_.sig))
+
+  def getUnfinalized: F[Set[Signed[DeployData]]] =
```

Comment:
This is effectively the whole content of deploy storage right? I wonder why call it `Unfinalized`. It just happens that we are not removing deploys right after we put it in block and wait for finalization to remove, because they are for sure in main chain. I mean - there are no `finalized` deploys in deploy storage

Reply:
Yes, absolutely.